Add support for Managed Instance for Apache Cassandra

frrossms opened the pull request:
Based on Swagger PRs: 
- https://github.com/Azure/azure-rest-api-specs/pull/12861 (has Approved-BreakingChange label)
- https://github.com/Azure/azure-rest-api-specs/pull/13244 (bugfix that slipped through review above)

# All SDK Contribution checklist:

This checklist is used to make sure that common guidelines for a pull request are followed.
- [x] **Please open PR in `Draft` mode if it is:**
	- Work in progress or not intended to be merged.
	- Encountering multiple pipeline failures and working on fixes.
- [x] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/master/CONTRIBUTING.md).**
- [x] **The pull request does not introduce [breaking changes](https://github.com/dotnet/corefx/blob/master/Documentation/coding-guidelines/breaking-change-rules.md).**

### [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-net/blob/master/CONTRIBUTING.md#general-guidelines)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/master/CONTRIBUTING.md#testing-guidelines)
- [x] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/master/CONTRIBUTING.md#sdk-generation-guidelines)
- [x] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code. (Track 2 only)
- [x] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK. Please double check nuget.org current release version.

## Additional management plane SDK specific contribution checklist: 
Note: Only applies to `Microsoft.Azure.Management.[RP]` or `Azure.ResourceManager.[RP]`
 
- [x] Include updated [management metadata](https://github.com/Azure/azure-sdk-for-net/tree/master/eng/mgmt/mgmtmetadata).
- [x] Update AzureRP.props to add/remove version info to maintain up to date API versions.

### Management plane SDK Troubleshooting
- If this is very first SDK for a services and you are adding new service folders directly under /SDK, please add `new service` label and/or contact assigned reviewer.
- If the check fails at the `Verify Code Generation` step, please ensure:
	- Do not modify any code in generated folders.
	- Do not selectively include/remove generated files in the PR.
	- Do use `generate.ps1/cmd` to generate this PR instead of calling `autorest` directly.
	Please pay attention to the @microsoft.csharp version output after running generate.ps1. If it is lower than current released version (2.3.82), please run it again as it should pull down the latest version,

### Old outstanding PR cleanup
 Please note:
	If PRs (including draft) has been out for more than 60 days and there are no responses from our query or followups, they will be closed to maintain a concise list for our reviewers.

frrossms committed:
Generate C# from Swagger.

Ran the command:

autorest C:\Users\frross\source\repos\azure-rest-api-specs\specification\cosmos-db\resource-manager\readme.md `
    --csharp --version=v2 --reflect-api-versions `
    --csharp-sdks-folder=C:\Users\frross\source\repos\azure-sdk-for-net\sdk `
    --tags=package-2021-03-preview

frrossms committed:
Copied test files from 2020-06-01-preview release of SDK.

The preview release has quite a different structure, so I went back to the last release from it (commit 18e1b965aa9de01c73d3cbe701b755cf4f8f2daa) and copied the test files here. The project builds, but the tests have a couple errors to fix.

frrossms committed:
Merged old tests with newer versions so they pass.

frrossms committed:
Record new test sessions for playback.

frrossms committed:
Update versions in metadata.

frrossms committed:
Add CRUD test for managed Cassandra instances.

frrossms committed:
Update autorest metadata.

The Swagger changes needed were checked in, so rerun autorun from the master branch in the main repo instead of against a local checkout so the metadata is correct.

m-nash commented on sdk/cosmosdb/Microsoft.Azure.Management.CosmosDB/src/Generated/CassandraResourcesOperations.cs:
```diff
@@ -2790,13 +2790,14 @@ public async Task<AzureOperationResponse<ThroughputSettingsGetResults>> BeginMig
             if ((int)_statusCode != 200 && (int)_statusCode != 202)
             {
-                var ex = new ErrorResponseUpdatedFormatException(string.Format("Operation returned an invalid status code '{0}'", _statusCode));
+                var ex = new CloudException(string.Format("Operation returned an invalid status code '{0}'", _statusCode));
```
This will introduce a breaking change.  Please have the breaking change team add the Approved-BreakingChange label to the spec PR https://github.com/Azure/azure-rest-api-specs/pull/13244

frrossms replied:
Oh no, not again. This is a place where our preview and release Swagger versions are out of sync. I guess I'll go force it back to being an ErrorResponseUpdatedFormat and regenerate the C#.

frrossms replied:
I should have pointed you to https://github.com/Azure/azure-rest-api-specs/pull/12861, which was the previous PR (a bug slipped through and had to get fixed). That PR has the Approved-BreakingChange label on it.

frrossms commented:
@m-nash I'm getting test failures saying it can't find an ARM template from TestData that is supposed to be copied to the output directory. I got that pattern from other parts of the repo that do the same thing. Do you have any insight on why it's not working on Github's tests?

frrossms committed:
Fix tests that were breaking on macOS and Linux.

I had hardcoded a \ instead of using Path.PathSeparator.

frrossms committed:
Try another way of fixing build errors in CI

Instead of Path.Separator, which is making the build fail, just use a /, which is supported on all platforms.

m-nash reviewed (approved):
Approved since breaking change approved label was added

MehaKaushik commented on sdk/cosmosdb/Microsoft.Azure.Management.CosmosDB/src/Microsoft.Azure.Management.CosmosDB.csproj:
```diff
@@ -6,17 +6,14 @@
     <PackageId>Microsoft.Azure.Management.CosmosDB</PackageId>
     <Description>Provides developers with libraries for the CosmosDB under Azure Resource manager to manage CosmosDB Account, Databases and child resources and their available management capabilities. Create, Delete, Update CosmosDB Account, Databases and child resources. Note: This client library is for CosmosDB under Azure Resource Manager.</Description>
-    <Version>2.0.0</Version>
+    <Version>2.1.0</Version>
```
this version is wrong. It should be 2.1.0-preview. I cannot release without this change.
